relay-server/src/metrics_extraction/generic.rs

```diff
@@ -56,13 +56,20 @@ where
             continue;
         };
 
+        // For extracted metrics we assume the `received_at` timestamp is equivalent to the time
+        // in which the metric is extracted.
+        #[cfg(not(test))]
```

Comment:
Instead of mocking time, we generate it differently if we are in test, in order to make the unit tests deterministic.